sky/utils/kubernetes/kind_port_config.py

```diff
@@ -0,0 +1,112 @@
+    """Set the `kubernetes:ports` to be `ingress` and backup original value.
+
+    If there is currently no config file, we then create the ~/.sky/config.yaml
+    file and set the `kubernetes:ports` value to `loadbalancer`. We then store
+    the current value (either `ingress` or `loadbalancer`) to the backup file
+    at ~/.sky/backup_config.yaml, and finally set the value to `ingress` in
+    the active config file.
+    """
+
+    if not skypilot_config.loaded_config_path():
+        logger.debug(
+            f'Making base {CONFIG_PATH} file as it did not previously exist.')
+        default_config = {
+            'kubernetes': {
+                'ports': kubernetes_enums.KubernetesPortMode.LOADBALANCER.value
+            }
+        }
+        config_path = os.path.expanduser(CONFIG_PATH)
+        common_utils.dump_yaml(config_path, default_config)
+        skypilot_config.try_load_config()
+
+    logger.debug(f'Creating backup at {BACKUP_PATH} of'
+                 'current kubernetes port forwarding option.')
+    backup_path = os.path.expanduser(BACKUP_PATH)
+    backup_port_config = skypilot_config.get_nested(
+        ('kubernetes', 'ports'),
+        kubernetes_enums.KubernetesPortMode.LOADBALANCER.value)
+    common_utils.dump_yaml(backup_path,
+                           {'kubernetes': {
+                               'ports': backup_port_config
+                           }})
+
+    current_path = skypilot_config.loaded_config_path()
+    logger.debug(
+        f'Set current port forwarding option to ingress in {current_path}.')
+    updated_config = skypilot_config.set_nested(
+        ('kubernetes', 'ports'),
+        kubernetes_enums.KubernetesPortMode.INGRESS.value)
+    common_utils.dump_yaml(current_path, updated_config)
+    skypilot_config.try_load_config()
+
+
+def restore_port_config():
+    """Restore the original value of `kubernetes:ports`.
+
+    Restores the value of `kubernetes:ports` from ~/.sky/backup_config.yaml
+    in the active config file. Then, clears the ~/.sky/backup_config.yaml
+    file.
+    """
+
+    backup_path = os.path.expanduser(BACKUP_PATH)
+    try:
+        backup_port_config = common_utils.read_yaml(
+            backup_path)['kubernetes']['ports']
+    except FileNotFoundError:
+        logger.debug(f'Could not find backup file at {BACKUP_PATH}.')
+        return
+    except (TypeError, KeyError):
+        logger.debug(f'Could not find `kubernetes:ports` in {BACKUP_PATH}.')
+        return
+
+    current_path = skypilot_config.loaded_config_path()
+    if current_path is None:
+        logger.debug('There is no current active config file.')
+        return
+
+    logger.debug(f'Restore original port forwarding option in {current_path}'
+                 f'from {BACKUP_PATH}.')
+    restored_config = skypilot_config.set_nested(('kubernetes', 'ports'),
+                                                 backup_port_config)
+    common_utils.dump_yaml(current_path, restored_config)
```

Comment:
Would this preserve any commented fields in my `config.yaml`? I.e., if I have some comments in my config.yaml and I run `sky local up` then `sky local down`, would my commented fields still exist? If they would be lost, that is a little concerning..